alerting-config-service-impl/src/main/java/org/hypertrace/alerting/config/service/EventConditionConfigServiceImpl.java

```diff
@@ -100,7 +101,9 @@ public void deleteEventCondition(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       requestValidator.validateDeleteEventConditionRequest(requestContext, request);
-      eventConditionStore.deleteObject(requestContext, request.getEventConditionId());
+      eventConditionStore
+          .deleteObject(requestContext, request.getEventConditionId())
+          .orElseThrow(Status.NOT_FOUND::asRuntimeException);
```

Comment:
so need to be careful now to go through all usages and change them to throw, since we don't return on delete generally the compiler won't catch the change - this is why I felt throwing in the delete case vs returning the optional was more suited in delete (while get does return, so the optional there felt more natural).

Anyway, we have both versions now so easy to go with either one - what do you think @saxenakshitiz ?

Reply:
In my view, it is fine to return an optional. It is better that caller make the decision whether to return not_found status or just return nothing.